saleor/static/dashboard-next/AppRoot.tsx

```diff
@@ -92,22 +188,96 @@ const ResponsiveDrawer = decorate<ResponsiveDrawerProps>(
   )
 );
 
+interface IMenuItem {
+  ariaLabel: string;
+  children?: IMenuItem[];
+  icon: React.ReactNode;
+  label: string;
+  url?: string;
+}
+interface MenuListProps {
+  menuItems: IMenuItem[];
+  nested?: boolean;
+  onMenuItemClick: (url: string) => void;
+}
+const MenuList = decorate<MenuListProps>(
+  ({ classes, menuItems, nested, onMenuItemClick }) => (
+    <div className={nested ? classes.menuListNested : classes.menuList}>
+      {menuItems.map(menuItem => {
+        if (!menuItem.url) {
+          return (
+            <Toggle key={menuItem.label}>
+              {(openedMenu, { toggle: toggleMenu }) => (
+                <>
+                  <div onClick={toggleMenu} className={classes.menuListItem}>
+                    {menuItem.icon}
+                    <Typography
+                      aria-label={menuItem.ariaLabel}
+                      className={classes.menuListItemText}
+                    >
+                      {menuItem.label}
+                    </Typography>
+                  </div>
+                  {openedMenu && (
+                    <MenuList
+                      menuItems={menuItem.children}
+                      nested={true}
```

Comment:
Please refactor this so the nesting is done by the parent and not the child. Have top-level menu positioned by its parent.